File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/invocation/entity/DubboInvocation.java
package org.javamaster.invocationlab.admin.service.invocation.entity;

import org.javamaster.invocationlab.admin.service.creation.entity.RequestParam;
import org.javamaster.invocationlab.admin.service.invocation.Invocation;

import java.util.List;

/**
 * @author yudong
 */
public class DubboInvocation implements Invocation {
    String javaMethodName;
    List<RequestParam> requestParams;

    @Override
    public String getJavaMethodName() {
        return this.javaMethodName;
    }

    public void setJavaMethodName(String javaMethodName) {
        this.javaMethodName = javaMethodName;
    }

    @Override
    public List<RequestParam> getParams() {
        return this.requestParams;
    }

    public void setRequestParams(List<RequestParam> requestParams) {
        this.requestParams = requestParams;
    }
}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/invocation/entity/DubboInvocation.kt
package org.javamaster.invocationlab.admin.service.invocation.entity

import org.javamaster.invocationlab.admin.service.creation.entity.RequestParam
import org.javamaster.invocationlab.admin.service.invocation.Invocation

/**
 * @author yudong
 */
class DubboInvocation : Invocation {
    private var javaMethodName: String? = null
    private var requestParams: List<RequestParam>? = null

    override fun getJavaMethodName(): String {
        return javaMethodName!!
    }

    override fun setJavaMethodName(javaMethodName: String) {
        this.javaMethodName = javaMethodName
    }

    override fun getParams(): List<RequestParam> {
        return requestParams!!
    }

    override fun setRequestParams(requestParams: List<RequestParam>) {
        this.requestParams = requestParams
    }
}
